src/relay/op/nn/sparse.cc

```diff
@@ -93,5 +93,51 @@ RELAY_REGISTER_OP("nn.sparse_dense")
     .set_support_level(1)
     .add_type_rel("SparseDense", SparseDenseRel);
 
+// relay.nn.sparse_transpose
+TVM_REGISTER_NODE_TYPE(SparseTransposeAttrs);
+
+bool SparseTransposeRel(const Array<Type>& types, int num_inputs, const Attrs& attrs,
+                    const TypeReporter& reporter) {
+  CHECK_EQ(types.size(), 4);
+  const auto* sparse_data = types[0].as<TensorTypeNode>();
+  CHECK_EQ(sparse_data->shape.size(), 1);
+  const auto* sparse_indices = types[1].as<TensorTypeNode>();
+  CHECK_EQ(sparse_indices->shape.size(), 1);
+  const auto* sparse_indptr = types[2].as<TensorTypeNode>();
+
+  std::vector<Type> output_types;
+  output_types.push_back(TensorTypeNode::make(sparse_data->shape, sparse_data->dtype));
+  output_types.push_back(TensorTypeNode::make(sparse_indices->shape, sparse_indices->dtype));
+  output_types.push_back(TensorTypeNode::make(sparse_indptr->shape, sparse_indptr->dtype));
+
+  reporter->Assign(types[3], TupleTypeNode::make(Array<Type>(output_types)));
+  return true;
+}
+
+Expr MakeSparseTranspose(Expr sparse_data, Expr sparse_indices, Expr sparse_indptr) {
+  auto attrs = make_node<SparseTransposeAttrs>();
+  static const Op& op = Op::Get("nn.sparse_transpose");
+  return CallNode::make(op, {sparse_data, sparse_indices, sparse_indptr}, Attrs(attrs), {});
+}
+
+TVM_REGISTER_API("relay.op.nn._make.sparse_transpose")
+.set_body_typed(MakeSparseTranspose);
+
+
+RELAY_REGISTER_OP("nn.sparse_transpose")
+    .describe(R"code(Transpose a sparse matrix X. Only support square sparse matrix
+
+- **input**: `(N, N)`
+- **out**: `(N, N)`.
+
+)code" TVM_ADD_FILELINE)
+    .set_attrs_type_key("relay.attrs.SparseTransposeAttrs")
+    .set_num_inputs(3)
+    .add_argument("sparse_data", "1D Tensor", "Sparse data matrix.")
+    .add_argument("sparse_indices", "1D Tensor", "Sparse indices matrix.")
+    .add_argument("sparse_indptr", "1D Tensor", "Sparse indptr matrix.")
```

Comment:
Spell out index pointer instead of indptr

Reply:
Resolved